src/meta/src/stream/stream_manager.rs

```diff
@@ -121,21 +174,116 @@ where
-            .await
-        {
-            for revert_func in revert_funcs.into_iter().rev() {
-                revert_func.await;
+        let table_id = table_fragments.table_id();
+        let (sender, mut receiver) = tokio::sync::mpsc::channel(10);
+        let execution = StreamingJobExecution::new(table_id, sender.clone());
+        self.creating_job_info.add_job(execution).await;
+
+        let stream_manager = self.clone();
+        tokio::spawn(async move {
+            let mut revert_funcs = vec![];
+            let res = stream_manager
+                .create_streaming_job_impl(&mut revert_funcs, table_fragments, ctx)
+                .await;
+            match res {
+                Ok(_) => {
+                    let _ = sender
+                        .send(CreatingState::Created)
+                        .await
+                        .inspect_err(|_| tracing::warn!("failed to notify created: {table_id}"));
+                }
+                Err(err) => {
+                    for revert_func in revert_funcs.into_iter().rev() {
+                        revert_func.await;
+                    }
+                    let _ = sender
+                        .send(CreatingState::Failed {
+                            reason: err.clone(),
+                        })
+                        .await
+                        .inspect_err(|_| {
+                            tracing::warn!("failed to notify failed: {table_id}, err: {err}")
+                        });
+                }
             }
-            return Err(e);
-        }
-        Ok(())
+        });
+
+        let res = try {
+            while let Some(state) = receiver.recv().await {
+                match state {
+                    CreatingState::Failed { reason } => {
+                        return Err(reason);
+                    }
+                    CreatingState::Canceling => {
+                        if let Ok(table_fragments) = self
+                            .fragment_manager
+                            .select_table_fragments_by_table_id(&table_id)
+                            .await
+                        {
+                            // try to cancel buffered creating command.
+                            if self
+                                .barrier_scheduler
+                                .try_cancel_scheduled_create(table_id)
+                                .await
+                            {
+                                tracing::debug!(
+                                    "cancelling streaming job {table_id} in buffer queue."
+                                );
+                                let node_actors = table_fragments.worker_actor_ids();
+                                let cluster_info =
+                                    self.cluster_manager.get_streaming_cluster_info().await;
+                                let node_actors = node_actors
```

Comment:
Seems this code snippet has occurred multiple times. 🥵 Is it possible to extract it, like `CommandContext::cleanup`?

Reply:
`CommandContext` only works for scheduled command in barrier manager, in this scenario the command will be simply removed from the queue. 🥵 There're three places that contains the same code snippet, have not yet decided which place is more suitable for this logic.

Reply:
Added a `clean_up` function for cancel/drop/reschedule commands in `CommandContext`. And just keep the code snippet here for this scenario.